12-watchtower-API.md

```diff
@@ -2,247 +2,337 @@
-- ```dispute_delta```: (Relative) Block number
-- ```transaction_size```: Measured in Bytes (e.g. 200) 
-- ```transaction_fee```: Measured in sats (e.g. 2000)
-- ```cipher```: AESGCM256, CHACHA20
-- ```customer_signature_algorithm```: ECDSA, SCHNORR
+* MUST fail the connection if:
+	* Any of the fields is missing.
+	* `receipt` does not matches the format specified at 	[Receipt-Format](#receipt-format)
+	* `receipt` fields do not match the ones sent in the `appointment` message.
+	* `wt_signature_algorithm` does not match any of the ones offered by the server.
+	* `wt_signature` cannot be verified using `wt_public_key`.
 
-### Rationale
+#### Receipt Format 
 
-We can group the data fields into logical groups. 
+The server MUST create the receipt containing the following information:
 
-* **Appointment information**: The appointment time is defined using the ```start_block``` and```end_block```. WatchTower will delete the job when the appointment has expired. We recommend only using block numbers as that is the natural clock for Bitcoin. 
-* **Explicit acknowledgement of transaction details**: The ```dispute_delta```, ```transaction_size```, and ```transaction_fee``` let the WatchTower confirm the transaction is "reasonable" and it can be accepted into the blockchain (especially if there is congestion in the future). 
-* **Encrypted transaction**: The ```cipher```,```encrypted_blob``` states how the WatchTower can later find the dispute transaction and decrypt the justice transaction. 
-* **Customer signature** The ```customer_public_key``` and ```customer_signature``` provides an explicit message about the job from the customer. 
+	txlocator
+	start_block
+	end_block
+	dispute_delta
+	encrypted_blob
+	transaction_size
+	transaction_fee
+	cipher
+	customer_signature
+	wt_public_key
+	
+
+#### Rationale
 
-Generally, this standard is trying to allow a reputationally accountable watching service. The signed job from the customer provides an explicit acknowledgement of the transaction details that is important for the WatchTower to decide whether they can accept it. If the decrypted justice transaction does not satisfy the signed job (e.g. fee too low), then the WatchTower is not obliged to fulfil it. 
+We assume the client has a well-known public key for the WatchTower. 
 
-The _explictiness_ of the signed job ensures there is a clear protocol trasncript between the customer and WatchTower. Given the blockchain and decrypted justice transaction, anyone can verify that the WatchTower could have satisified the job. 
+The receipt contains, mainly, the information provided by the user. The WatchTower will need to sign the receipt to provide evidence of agreement.
+
+The `customer_signature` is included in the receipt to link both the client request and the server response. Otherwise a client could sign random data and claim to have send it to the tower. In the same way, a signed receipt protects the user from false claims from the tower.
```

Comment:
I thought we were getting rid of the customer signature? Reading the BOLT, the auth_token already performs the authentication from client to server. 

AFAIK, the only purpose for the client receipt, is that PISA can also sign it, and then anyone can verify that PISA seen the customer's request and this receipt correspond to that. 

But is it worth the implementation head-ache of getting that to work? It would mean the customer has to register a key with us which defeats purpose of a blinded token approach?

Reply:
That's exactly what is said there, that the customer signature is included to link both request and response.

The user does not have to register anything with us, they can use an ephemeral public key (use only once). The only time where the key may be linked to someone is if they're issuing a claim, and that makes complete sense.

Not implementing this defuses the accountability bit since gives the tower a straight forward way of not providing a service to the user and gives the user no evidence of it.

Reply:
the user always has a signed receipt that the watch tower accepted the job. so the watch tower cant pretend it never got it?

i thought the issue was to protect the watch tower? i.e. it issued receipts according to the job request from the customer?

Reply:
The tower can sign a receipt with different data that the one the user sent her, and the user has no way to prove whether that's true or not. If the tower has to sign the customer signature there is evidence of what the user sent since the tower cannot forge that signature.

Reply:
Put that into the bolt then. that sounds more convincing.

although we could also make this optional. i.e. maybe some wallet developers dont want to do it.

Reply:
An optional part of an optional extension? I personally think that's too many level of optionality. Again the purpose of the whole draft is to raise discussion so I'm happy considering all those things once we get feedback.

Reply:
I'll add the rationale in though.